src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RSA.cs

```diff
@@ -293,139 +288,75 @@ public virtual bool VerifyData(ReadOnlySpan<byte> data, ReadOnlySpan<byte> signa
 
         public virtual byte[] ExportRSAPrivateKey()
         {
-            using (AsnWriter pkcs1PrivateKey = WritePkcs1PrivateKey())
+            RSAParameters rsaParameters = ExportParameters(true);
+
+            try
+            {
+                using (AsnWriter pkcs1PrivateKey = RSAKeyFormatHelper.WritePkcs1PrivateKey(rsaParameters))
+                {
+                    return pkcs1PrivateKey.Encode();
+                }
+            }
+            finally
             {
-                return pkcs1PrivateKey.Encode();
+                ClearPrivateParameters(rsaParameters);
             }
         }
 
         public virtual bool TryExportRSAPrivateKey(Span<byte> destination, out int bytesWritten)
         {
-            using (AsnWriter pkcs1PrivateKey = WritePkcs1PrivateKey())
+            RSAParameters rsaParameters = ExportParameters(true);
+
```

Comment:
All of the pinning that was done in `WritePkcs1PrivateKey` now needs to be done here.